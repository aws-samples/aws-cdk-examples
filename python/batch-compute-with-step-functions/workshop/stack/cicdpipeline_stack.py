from aws_cdk import (
    core
)

from construct.cicdpipeline.cicd_batch import CICDBatch
from construct.cicdpipeline.cicd_web import CICDWeb

class CICDPipelineStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str,
        UserName="default",
        EmailAddress="default",
        BatchRepo="default",
        WebRepo="default",
        WebService="default",
        **kwargs
    ) -> None:
        super().__init__(scope, id, **kwargs)
        
        self.My_CICDBatch = CICDBatch(self,
            "CICDBatch-" + UserName,
            UserName=UserName,
            Repo=BatchRepo
        )
        
        self.My_CICDWeb = CICDWeb(self,
            "CICDWeb-" + UserName,
            UserName=UserName,
            Repo=WebRepo,
            WebService=WebService
        )