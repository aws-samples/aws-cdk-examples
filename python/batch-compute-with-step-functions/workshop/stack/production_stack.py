from aws_cdk import (
    aws_stepfunctions as _sfn,
    aws_stepfunctions_tasks as _sfn_tasks,
    aws_ec2 as _ec2,
    aws_s3 as _s3,
    aws_iam as _iam,
    aws_apigateway as _apigateway,
    core,
)
from construct.production.batch_env import BatchENV
from construct.production.stepfunctions_env import StepfunctionsENV
from construct.production.sns_env import SnsENV
from construct.production.batch_task import BatchTASK
from construct.production.lambda_task import LambdaTask
from construct.production.ecr_env import EcrENV

class ProductionStack(core.Stack):
    def getStateMachine(self):
        return self.My_SFN.statemachine
        
    def getEcrRepo(self):
        return self.My_ECR_Repo
        
    def getVpc(self):
        return self.My_Vpc

    def __init__(self, scope: core.Construct, id: str,UserName="default",EmailAddress="default",**kwargs) -> None:
        super().__init__(scope, id, **kwargs)
        
        # vpc with one public subnet and one private subnet
        self.My_Vpc = _ec2.Vpc(self, "vpc-"+ UserName + "-batch",
            max_azs=2,
            nat_gateways=1,
            subnet_configuration=[
                _ec2.SubnetConfiguration(
                    subnet_type=_ec2.SubnetType.PUBLIC,
                    name="BatchENV",
                    cidr_mask=24
                ),
                _ec2.SubnetConfiguration(
                    cidr_mask=24,
                    name="InternalENV",
                    subnet_type=_ec2.SubnetType.PRIVATE
                )
            ]
        )
        
        # Definition Of S3 Bucket For Batch Computing
        self.My_S3_Bucket = _s3.Bucket(self,
            "s3bucket-" + UserName + "-batch",
            lifecycle_rules=[
                _s3.LifecycleRule(
                    # delete the files after 1800 days (5 years)
                    expiration=core.Duration.days(365),
                    transitions=[
                        # move files into glacier after 90 days
                        _s3.Transition(
                            transition_after=core.Duration.days(30),
                            storage_class=_s3.StorageClass.GLACIER
                        ),
                        _s3.Transition(
                            transition_after=core.Duration.days(120),
                            storage_class=_s3.StorageClass.DEEP_ARCHIVE
                        )
                    ],
                )
            ],
            removal_policy=core.RemovalPolicy.DESTROY
        )
        
        # Definition Of ECR Repo
        self.My_ECR_Repo = EcrENV(self,
            "ecr-" + UserName + "-batch",
            UserName=UserName
        )
        
        # Definition Of Batch ENV For Batch Computing
        self.My_Batch = BatchENV(self,
            "env-" + UserName + "-batch",
            CurrentVPC=self.My_Vpc,
            TargetS3=self.My_S3_Bucket,
            UserName=UserName
            
        )
        
        # Definition Of Batch Job 
        self.My_Batch_Task = BatchTASK(self,
            "task-" + UserName + "-batch",
            EcrRepo=self.My_ECR_Repo,
            UserName=UserName
        )
        
        # Definition Of Lambda Job 
        self.My_Lambda_Task = LambdaTask(self,
            "task-" + UserName + "-lambda",
            TargetS3=self.My_S3_Bucket
        )
        
        # Definition Of SNS Topic With Subscription 
        self.My_SNS = SnsENV(self,
            "sns-" + UserName + "-sfn",
            UserName=UserName,
            EmailAddress=EmailAddress
        )
        
        # Definition Of State Machine In Step functions  
        self.My_SFN = StepfunctionsENV(self,
            "statemachine-" + UserName + "-sfn",
            QueueDefine = self.My_Batch,
            TaskDefine = self.My_Batch_Task,
            LambdaDefine = self.My_Lambda_Task,
            SNSDefine = self.My_SNS
        )
        
        core.CfnOutput(self,
            "S3 Bucket For AWS Batch",
            value = self.My_S3_Bucket.bucket_name
        )