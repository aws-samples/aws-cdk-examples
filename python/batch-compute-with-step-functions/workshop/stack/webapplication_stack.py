from aws_cdk import (
    aws_stepfunctions as _sfn,
    aws_stepfunctions_tasks as _sfn_tasks,
    aws_ec2 as _ec2,
    aws_s3 as _s3,
    aws_iam as _iam,
    aws_apigateway as _apigateway,
    core,
)

from construct.webapplication.apigateway import APIDefinition
from construct.webapplication.ecr_env import EcrENV
from construct.webapplication.ecs_env import EcsENV
from construct.webapplication.ecs_task import EcsTASK

class WebApplicationStack(core.Stack):
    
    def getEcrRepo(self):
        return self.My_ECR_Repo
    
    def getAPIGateway(self):
        return self.My_APIGW
        
    def getService(self):
        return self.My_ECS_Task

    def __init__(self, scope: core.Construct, id: str,
        UserName="default",
        EmailAddress="default",
        StateMachine="default",
        Vpc="default",
        **kwargs
    ) -> None:
        super().__init__(scope, id, **kwargs)
        
        self.My_ECR_Repo = EcrENV(self,
            "ecr-" + UserName + "-web",
            UserName=UserName
        )
        
        self.My_ECS_ENV = EcsENV(self,
            "ecs-" + UserName + "-web",
            UserName=UserName,
            Vpc=Vpc
        )
        
        self.My_APIGW = APIDefinition(self,
            "apigateway-" + UserName,
            UserName=UserName,
            StateMachine=StateMachine
        )
        
        self.My_ECS_Task = EcsTASK(self,
            "ecstask-" + UserName + "-web",
            UserName=UserName,
            Cluster=self.My_ECS_ENV.getEcsCluster("WebApplicationCluster"),
            Repo=self.My_ECR_Repo,
            Vpc=Vpc,
            APIGateway = self.My_APIGW
        )
        