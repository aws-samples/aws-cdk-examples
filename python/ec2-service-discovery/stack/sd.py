from aws_cdk import (
    aws_servicediscovery as sd,
    core,
)

class HttpApiServiceDiscovery(core.Stack):
    def __init__(self, app: core.App, id: str, props, **kwargs) -> None:
        super().__init__(app, id, **kwargs)

        namespace = sd.HttpNamespace(self, "httpns", name = props['namespace'])

        service = namespace.create_service("ip_service",
            description = "Service registering ip instances",
            custom_health_check = sd.HealthCheckCustomConfig(
                # The failure threshold indicates the number of 30-second intervals
                # you want AWS Cloud Map to wait between the time that your application
                # sends an UpdateInstanceCustomHealthStatus request
                failure_threshold = 1
            )
        )
        service.register_ip_instance("ip_instance", ipv4 = "54.239.25.192", port = 8080)

        core.CfnOutput(
            self, "SVCID",
            description = "SERVICE ID",
            value = service.service_id
        )

        self.output_props = props.copy()
        self.output_props['service']= service

    # pass objects to another stack
    @property
    def outputs(self):
        return self.output_props

class PrivateDnsServiceDiscovery(core.Stack):
    def __init__(self, app: core.App, id: str, props, **kwargs) -> None:
        super().__init__(app, id, **kwargs)

        namespace = sd.PrivateDnsNamespace(self, "privatedns", name = f"{props['namespace']}.local", vpc = props['vpc'])

        service = namespace.create_service("ip_service",
            dns_record_type = sd.DnsRecordType.A,
            dns_ttl = core.Duration.seconds(3),
            description = "Service registering ip instances",
        )

        core.CfnOutput(
            self, "SVCID",
            description = "SERVICE ID",
            value = service.service_id
        )

        self.output_props = props.copy()
        self.output_props['service']= service

    # pass objects to another stack
    @property
    def outputs(self):
        return self.output_props
