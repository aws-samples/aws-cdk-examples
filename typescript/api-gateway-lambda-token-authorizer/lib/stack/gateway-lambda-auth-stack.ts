import * as path from 'path'
import * as cdk from "aws-cdk-lib";
import { Construct } from "constructs";

/**
 * Stack, which creates LambdaRestApi Gateway, with TokenAuthorizer
 *
 * @export
 * @class GatewayLambdaAuth
 * @extends {cdk.Stack}
 */
export class GatewayLambdaAuth extends cdk.Stack {
  readonly operationalLambda: cdk.aws_lambda.IFunction;
  readonly lambdaIntegration: cdk.aws_apigateway.LambdaIntegration;

  readonly operationalEntryPath = path.join(__dirname + "/../lambdas/operational/index.ts")
  readonly authLambdaEntryPath = path.join(__dirname + "/../lambdas/authorizer/index.ts")

  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    /** Creating operational Lambda, which server the request */
    let operationalLambda = this.getOperationalFunction();

    /** Lambda, which takes incoming request and checks the authorization and authentication */
    let authorizerLambda = this.getLambdaAuthFunction();

    /** Generating Token Authorization, which will be injected to API Gateway */
    let lambdaTokenAuthorizer = this.getTokenAuthorizer(authorizerLambda)

    /** Creating Lambda Rest API, which integrates Endpoint to Lambda */
    let lambdaRestApi = this.createRestApi(operationalLambda, lambdaTokenAuthorizer);

    /** Creating /health resource at root for lambda Rest API */
    const healthResource = lambdaRestApi.root.addResource("health");
    healthResource.addMethod("GET");

    /** Returning Output with URL made as part of lambdaRestApi */
    new cdk.CfnOutput(this, "apiUrl", { value: lambdaRestApi.url });
  }


  /**
   * Creating Operational Lambda, to server the incoming request
   *
   * @private
   * @return {*}  {cdk.aws_lambda.IFunction}
   * @memberof GatewayLambdaAuth
   */
  private getOperationalFunction(): cdk.aws_lambda.IFunction {
    return new cdk.aws_lambda_nodejs.NodejsFunction(this, "operational-lambda", {
      runtime: cdk.aws_lambda.Runtime.NODEJS_18_X,
      handler: "index.handler",
      bundling: {
        sourceMap: true,
        minify: true,
      },
      description: 'Operational Lambda',
      entry: this.operationalEntryPath,
      environment: {
        NODE_OPTIONS: '--enable-source-maps',
      },
      logRetention: cdk.aws_logs.RetentionDays.ONE_DAY,
      memorySize: 512,
      timeout: cdk.Duration.minutes(2),
    });
  }


  /**
   * Creating Authorization Lambda, to validate incoming request
   *
   * @private
   * @return {*}  {cdk.aws_lambda.IFunction}
   * @memberof GatewayLambdaAuth
   */
  private getLambdaAuthFunction(): cdk.aws_lambda.IFunction {
    return new cdk.aws_lambda_nodejs.NodejsFunction(this, "authentication-lambda", {
      runtime: cdk.aws_lambda.Runtime.NODEJS_18_X,
      handler: "index.handler",
      bundling: {
        sourceMap: true,
        minify: true,
      },
      description: 'Lambda Authorizer',
      entry: this.authLambdaEntryPath,
      environment: {
        NODE_OPTIONS: '--enable-source-maps',
      },
      logRetention: cdk.aws_logs.RetentionDays.ONE_DAY,
      memorySize: 512,
      timeout: cdk.Duration.minutes(2),
    });
  }


  /**
   * Creating Token Authorizer, to inject auth-lambda into Rest API Gateway
   *
   * @private
   * @param {IFunction} authorizerLambda
   * @return {*}  {cdk.aws_apigateway.TokenAuthorizer}
   * @memberof GatewayLambdaAuth
   */
  private getTokenAuthorizer(authorizerLambda: cdk.aws_lambda.IFunction): cdk.aws_apigateway.TokenAuthorizer {
    return new cdk.aws_apigateway.TokenAuthorizer(
      this,
      "operationalAuthorizer",
      {
        handler: authorizerLambda,
      }
    );
  }


  /**
   * Creating Lambda Rest API, that integrates API to Operational Lambda with Token Authorizer
   *
   * @private
   * @param {cdk.aws_lambda.IFunction} operationalLambda
   * @param {cdk.aws_apigateway.TokenAuthorizer} lambdaAuthorizer
   * @return {*}  {cdk.aws_apigateway.LambdaRestApi}
   * @memberof GatewayLambdaAuth
   */
  private createRestApi(
    operationalLambda: cdk.aws_lambda.IFunction,
    lambdaAuthorizer: cdk.aws_apigateway.TokenAuthorizer
  ): cdk.aws_apigateway.LambdaRestApi {
    const logGroup = this.createApiGatewayAccessLogsGroup(this);
    return new cdk.aws_apigateway.LambdaRestApi(this, "rest-api-gateway", {
      handler: operationalLambda,
      proxy: false,
      deployOptions: {
        stageName: "dev",
        accessLogDestination: new cdk.aws_apigateway.LogGroupLogDestination(
          logGroup
        ),
        accessLogFormat:
          cdk.aws_apigateway.AccessLogFormat.jsonWithStandardFields(),
      },
      defaultMethodOptions: {
        authorizer: lambdaAuthorizer,
      },
    });
  }


  /**
   * Creating Access-log Group, for API Gateway
   *
   * @private
   * @param {Construct} scope
   * @return {*}  {cdk.aws_logs.ILogGroup}
   * @memberof GatewayLambdaAuth
   */
  private createApiGatewayAccessLogsGroup(
    scope: Construct
  ): cdk.aws_logs.ILogGroup {
    const logGroupName = "apigateway-auth-lambda";
    const logRetention = new cdk.aws_logs.LogRetention(
      scope,
      "apiGwLogGroupConstruct",
      {
        logGroupName: logGroupName,
        retention: cdk.aws_logs.RetentionDays.ONE_WEEK,
        removalPolicy: cdk.RemovalPolicy.DESTROY,
      }
    );
    const logGroup = cdk.aws_logs.LogGroup.fromLogGroupArn(
      scope,
      "apiGwLogGroup",
      logRetention.logGroupArn
    );
    return logGroup;
  }
}

/**
 *  API
 *  https://{GatewayId}.execute-api.us-east-1.amazonaws.com/dev/health
*/
