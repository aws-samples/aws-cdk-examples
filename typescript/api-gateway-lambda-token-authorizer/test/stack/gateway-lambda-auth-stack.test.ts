import { App } from "aws-cdk-lib";
import { GatewayLambdaAuth } from "../../lib/stack/gateway-lambda-auth-stack";
import { Template } from "aws-cdk-lib/assertions";

describe('Snapshot', () => {

    it('Stack', () => {
        const app = new App();
        const stack = new GatewayLambdaAuth(app, 'test-api-gateway-lambda-auth');
        const template = Template.fromStack(stack);
        expect(template.toJSON()).toMatchSnapshot();
    });

})

