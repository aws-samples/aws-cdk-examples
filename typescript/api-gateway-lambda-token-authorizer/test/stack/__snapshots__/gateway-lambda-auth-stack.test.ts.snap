// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Stack 1`] = `
{
  "Outputs": {
    "apiUrl": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "restapigatewayE22E31C5",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "restapigatewayDeploymentStagedevB80C9CD7",
            },
            "/",
          ],
        ],
      },
    },
    "restapigatewayEndpoint4F6651D0": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "restapigatewayE22E31C5",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "restapigatewayDeploymentStagedevB80C9CD7",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": {
      "DependsOn": [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "5bc602ecde93c947efe5899ae355f999986a1acbe610b1c0b9c468d738857555.zip",
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "logs:DeleteLogGroup",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:apigateway-auth-lambda:*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": [
          {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "apiGwLogGroupConstructA23954E8": {
      "Properties": {
        "LogGroupName": "apigateway-auth-lambda",
        "RemovalPolicy": "destroy",
        "RetentionInDays": 7,
        "ServiceToken": {
          "Fn::GetAtt": [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "authenticationlambdaDD3A2252": {
      "DependsOn": [
        "authenticationlambdaServiceRole9798A92B",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "a89b0c25113046ece0e7fb82e6d729811e22e9ed30e0d44feb9ffa39ef67b8a4.zip",
        },
        "Description": "Lambda Authorizer",
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "NODE_OPTIONS": "--enable-source-maps",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "authenticationlambdaServiceRole9798A92B",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "authenticationlambdaLogRetention32E73801": {
      "Properties": {
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "authenticationlambdaDD3A2252",
              },
            ],
          ],
        },
        "RetentionInDays": 1,
        "ServiceToken": {
          "Fn::GetAtt": [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "authenticationlambdaServiceRole9798A92B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "authenticationlambdatestapigatewaylambdaauthoperationalAuthorizer17083188Permissions72AE6440": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "authenticationlambdaDD3A2252",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "restapigatewayE22E31C5",
              },
              "/authorizers/",
              {
                "Ref": "operationalAuthorizer363A7D2B",
              },
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "operationalAuthorizer363A7D2B": {
      "Properties": {
        "AuthorizerUri": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Fn::Select": [
                  1,
                  {
                    "Fn::Split": [
                      ":",
                      {
                        "Fn::GetAtt": [
                          "authenticationlambdaDD3A2252",
                          "Arn",
                        ],
                      },
                    ],
                  },
                ],
              },
              ":apigateway:",
              {
                "Fn::Select": [
                  3,
                  {
                    "Fn::Split": [
                      ":",
                      {
                        "Fn::GetAtt": [
                          "authenticationlambdaDD3A2252",
                          "Arn",
                        ],
                      },
                    ],
                  },
                ],
              },
              ":lambda:path/2015-03-31/functions/",
              {
                "Fn::GetAtt": [
                  "authenticationlambdaDD3A2252",
                  "Arn",
                ],
              },
              "/invocations",
            ],
          ],
        },
        "IdentitySource": "method.request.header.Authorization",
        "Name": "testapigatewaylambdaauthoperationalAuthorizer17083188",
        "RestApiId": {
          "Ref": "restapigatewayE22E31C5",
        },
        "Type": "TOKEN",
      },
      "Type": "AWS::ApiGateway::Authorizer",
    },
    "operationallambdaFE43E13E": {
      "DependsOn": [
        "operationallambdaServiceRole14B56EA5",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "5ce4b859d5e57c3d238da3ac017e0ea37bf34ad1b71de268430b08e38bb9450d.zip",
        },
        "Description": "Operational Lambda",
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "NODE_OPTIONS": "--enable-source-maps",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "operationallambdaServiceRole14B56EA5",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "operationallambdaLogRetention43961A26": {
      "Properties": {
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "operationallambdaFE43E13E",
              },
            ],
          ],
        },
        "RetentionInDays": 1,
        "ServiceToken": {
          "Fn::GetAtt": [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "operationallambdaServiceRole14B56EA5": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "restapigatewayAccount4878E943": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "restapigatewayE22E31C5",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "restapigatewayCloudWatchRole0B08C582",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "restapigatewayCloudWatchRole0B08C582": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "restapigatewayDeploymentB29CB257e5bb083e30c50c343739c1961c3643d1": {
      "DependsOn": [
        "restapigatewayhealthGET9A80151C",
        "restapigatewayhealth0EB12846",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "restapigatewayE22E31C5",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "restapigatewayDeploymentStagedevB80C9CD7": {
      "DependsOn": [
        "restapigatewayAccount4878E943",
      ],
      "Properties": {
        "AccessLogSetting": {
          "DestinationArn": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":logs:",
                {
                  "Ref": "AWS::Region",
                },
                ":",
                {
                  "Ref": "AWS::AccountId",
                },
                ":log-group:",
                {
                  "Fn::GetAtt": [
                    "apiGwLogGroupConstructA23954E8",
                    "LogGroupName",
                  ],
                },
                ":*",
              ],
            ],
          },
          "Format": "{"requestId":"$context.requestId","ip":"$context.identity.sourceIp","user":"$context.identity.user","caller":"$context.identity.caller","requestTime":"$context.requestTime","httpMethod":"$context.httpMethod","resourcePath":"$context.resourcePath","status":"$context.status","protocol":"$context.protocol","responseLength":"$context.responseLength"}",
        },
        "DeploymentId": {
          "Ref": "restapigatewayDeploymentB29CB257e5bb083e30c50c343739c1961c3643d1",
        },
        "RestApiId": {
          "Ref": "restapigatewayE22E31C5",
        },
        "StageName": "dev",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "restapigatewayE22E31C5": {
      "Properties": {
        "Name": "rest-api-gateway",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "restapigatewayhealth0EB12846": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "restapigatewayE22E31C5",
            "RootResourceId",
          ],
        },
        "PathPart": "health",
        "RestApiId": {
          "Ref": "restapigatewayE22E31C5",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "restapigatewayhealthGET9A80151C": {
      "Properties": {
        "AuthorizationType": "CUSTOM",
        "AuthorizerId": {
          "Ref": "operationalAuthorizer363A7D2B",
        },
        "HttpMethod": "GET",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "operationallambdaFE43E13E",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "restapigatewayhealth0EB12846",
        },
        "RestApiId": {
          "Ref": "restapigatewayE22E31C5",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "restapigatewayhealthGETApiPermissionTesttestapigatewaylambdaauthrestapigateway1BA02674GEThealth4014F9D3": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "operationallambdaFE43E13E",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "restapigatewayE22E31C5",
              },
              "/test-invoke-stage/GET/health",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "restapigatewayhealthGETApiPermissiontestapigatewaylambdaauthrestapigateway1BA02674GEThealthD77C65E2": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "operationallambdaFE43E13E",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "restapigatewayE22E31C5",
              },
              "/",
              {
                "Ref": "restapigatewayDeploymentStagedevB80C9CD7",
              },
              "/GET/health",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
